Remove code gen

ambergristle opened the pull request:
### What changed
- removed code gen flows + logic
  - closes https://github.com/fiberplane/create-honc-app/issues/94 
- returned symbol prompt results from path + openapi prompts to handle cancellation
### Open questions
- description prompt was only used for `hatch` flow, but can still be used for `package.json` update
  - add as standard step, or remove altogether?
- previous update to path/project name logic removed use of target dir as project name
  - if no project name specified, should target dir name be used as project name?

brettimus commented on cli/src/actions/path.ts:
```diff
@@ -43,6 +43,11 @@ export async function promptPath(ctx: Context) {
         initialValue: false,
       });
 
+      if (typeof confirmation === "symbol") {
```
ah right i should've caught this! i was even looking at it at first being like "this can be a symbol, the boolean check should handle it"

brettimus commented on cli/src/index.ts:
```diff
@@ -32,17 +30,8 @@ async function main() {
-
   const prompts = [
-    shouldPromptDescription ? promptDescription : undefined,
+    // promptDescription,
```
```suggestion
```

brettimus commented on cli/src/index.ts:
```diff
@@ -7,14 +7,12 @@ import {
   promptDatabase,
 } from "./actions/database";
 import { actionDependencies, promptDependencies } from "./actions/dependencies";
-import { promptDescription } from "./actions/description";
```
could probably remove this file as well (`actions/description.ts`)

brettimus reviewed (approved):
appreuved with two nits! (dangling comment, removable file)

brettimus commented:
> if no project name specified, should target dir name be used as project name?

you mean in the package.json? yes that seems reasonable. not sure if we need to transform it at all for use in a package.json but i think repurposing the dirname is a good default

ambergristle commented:
@brettimus removed the `description` prompt + files, and replaced with a stock message:
````A lightweight Hono backend with a ${upperFirst(context.database)} database````

also added a fall-back to dir name in case project name is left unset
